### PR TITLE
fix(router): reject invalid app route discovery conflicts

### PR DESCRIPTION
**Pull request by NathanDrake2406**
## Summary

- Reject App Router interception routes that use `( .. )`-style upward markers after the normalized route has no visible segment left to climb.
- Report an explicit page/route handler conflict when `page` and `route` files resolve to the same App Router path.
- This intentionally groups two scanner-time validation/DX parity issues in one PR, while leaving slot ownership behavior for a separate PR.

## Root cause / Why

vinext already resolved interception targets by visible route segments, but it kept resolving at root when `( .. )` or `( .. )( .. )` tried to climb past the normalized intercepting route. Next.js validates the normalized interception route first and throws explicit marker errors instead of silently resolving from root.

vinext also discovered `page` and `route` files as separate entries and deferred conflict detection to the generic duplicate route pattern validator. That produced a correct rejection but not the explicit App Router conflict that points at the page and route handler files.

Next.js parity references:

- [Interception route validation source](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/packages/next/src/shared/lib/router/utils/interception-routes.ts#L60-L95)
- [Interception route validation tests](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/packages/next/src/shared/lib/router/utils/interception-routes.test.ts#L66-L75)
- [Route handler docs: page.js and route.js conflict](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/docs/01-app/01-getting-started/15-route-handlers.mdx#L150-L163)
- [App path normalization source](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/packages/next/src/shared/lib/router/utils/app-paths.ts#L23-L51)

## Verification

- `vp test run tests/routing.test.ts`
- `vp test run tests/route-sorting.test.ts`
- `vp check tests/routing.test.ts packages/vinext/src/routing/app-router.ts`

### CODE REVIEW

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and test gaps.

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and whether the tests cover the real behavior and meaningful edge cases.

File: tests/routing.test.ts [hunk 1]
```diff
@@ -610,6 +626,69 @@ describe("appRouter - route discovery", () => {
...
+      await writeFile(path.join(appDir, "(group)", "@modal", "(..)foo", "page.tsx"), EMPTY_PAGE);
+
+      invalidateAppRouteCache();
+      await expect(appRouter(appDir)).rejects.toThrow(
+        "Cannot use (..) marker at the root level, use (.) instead.",
+      );
+    });
+  });
+
+  it("rejects (..)(..) intercepting routes at root or one visible level up", async () => {
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply tests/routing.test.ts:663]: Addressed in 40d91fd. Added a positive route-group test for `app/shop/(group)/@modal/(..)foo/page.tsx`, confirming `( .. )` remains valid when the normalized intercepting route has one visible parent segment and resolves to `/foo`.

File: packages/vinext/src/routing/app-router.ts [hunk 2]
```diff
@@ -1035,6 +1092,36 @@ function computeInterceptTarget(
...
 
+function formatInterceptionRoutePath(
+  routeSegments: string[],
+  convention: string,
+  interceptSegment: string,
+  nestedParts: string[],
+): string {
+  const marker = markerForInterceptionConvention(convention);
+  const convertedRoute = convertSegmentsToRouteParts(routeSegments);
+  const prefix = convertedRoute ? convertedRoute.urlSegments : [];
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/routing/app-router.ts]: Addressed in 40d91fd. `formatInterceptionRoutePath` now falls back to visible raw route segments when conversion fails, so malformed route diagnostics keep the surrounding route context.

File: packages/vinext/src/routing/app-router.ts [hunk 3]
```diff
@@ -239,6 +240,46 @@ export async function appRouter(
...
+
+    if (!entry.pagePath && route.pagePath) {
+      entry.pagePath = route.pagePath;
+    }
+    if (!entry.routePath && route.routePath) {
+      entry.routePath = route.routePath;
+    }
+  }
+
+  for (const [pattern, entry] of byPattern) {
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/routing/app-router.ts:266]: Addressed in 40d91fd. Added a comment explaining that this validation intentionally runs before generic duplicate-pattern validation so the page/route conflict error can name both files.

File: packages/vinext/src/routing/app-router.ts [hunk 4]
```diff
@@ -1012,6 +1053,22 @@ function computeInterceptTarget(
...
         }
       }
+      if (climbed < levelsToClimb) {
+        const interceptionRoute = formatInterceptionRoutePath(
+          routeSegments,
+          convention,
+          interceptSegment,
+          path.relative(interceptRoot, currentDir).split(path.sep).filter(Boolean),
+        );
+        if (convention === "..") {
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/routing/app-router.ts:1065]: Checked this one and left the branching as-is. The current `case` only handles the two upward conventions, and the explicit `(..)` branch plus default `(..)(..)` error matches the supported marker set without adding another abstraction.

File: packages/vinext/src/routing/app-router.ts [hunk 5]
```diff
@@ -239,6 +240,48 @@ export async function appRouter(
...
+      });
+      continue;
+    }
+
+    if (!entry.pagePath && route.pagePath) {
+      entry.pagePath = route.pagePath;
+    }
+    if (!entry.routePath && route.routePath) {
+      entry.routePath = route.routePath;
+    }
```
Comment (james-elicx) [inline packages/vinext/src/routing/app-router.ts:263]: nit: assuming empty strings aren't a thing, i think you could also do something like 
```ts
entry.pagePath ||= route.pagePath;
entry.routePath ||= route.routePath;
```

File: packages/vinext/src/routing/app-router.ts [hunk 6]
```diff
@@ -239,6 +240,48 @@ export async function appRouter(
...
+
+  for (const [pattern, entry] of byPattern) {
+    if (!entry.pagePath || !entry.routePath) continue;
+
+    throw new Error(
+      `Conflicting route and page at ${pattern}: route at ${formatAppFilePath(
+        entry.routePath,
+        appDir,
+      )} and page at ${formatAppFilePath(entry.pagePath, appDir)}`,
+    );
```
Comment (james-elicx) [inline packages/vinext/src/routing/app-router.ts:274]: nit: it might be nicer for the user if we collected a list of all the conflicts to throw in the error, rather than the first conflict for a pattern.
